### PR TITLE
BigQuery: many new samples

### PR DESCRIPTION
**Pull request by shollyman**
Introduces a variety of sample snippets across the product.  ORC format, Parquet format, CMEK, schema manipulations.

Also, wraps common include block used by all samples in a no-op function, to ensure that samples have a consistent indent depth.

### CODE REVIEW

**Comment by shollyman** [comment]
I forgot this was dependent on some changes to the client library that I haven't submitted yet.  Will ping this once those changes are in.

**Comment by shollyman** [comment]
Dependent changes went in with https://code-review.googlesource.com/c/gocloud/+/31670

File: bigquery/snippets/snippet.go [hunk 1]
```diff
@@ -16,115 +16,161 @@ import (
...
-// [END bigquery_update_dataset_access]
-// [END bigquery_update_dataset_description]
-// [END bigquery_update_dataset_expiration]
-// [END bigquery_update_table_description]
-// [END bigquery_update_table_expiration]
+func noOpCommentFunc() {
+	// Use a common block to inline comments related to importing the library
+	// and constructing a client.  Inside a func to ensure the indentation is
+	// consistent between multiple includes.
+	// [START bigquery_add_empty_column]
```
Comment (tswast) [inline bigquery/snippets/snippet.go]: 👍 on alphabetizing this.

File: bigquery/snippets/snippet.go [hunk 2]
```diff
@@ -407,6 +453,194 @@ func createTableExplicitSchema(client *bigquery.Client, datasetID, tableID strin
...
+				{Name: "city", Type: bigquery.StringFieldType},
+				{Name: "state", Type: bigquery.StringFieldType},
+				{Name: "zip", Type: bigquery.StringFieldType},
+				{Name: "numberOfYears", Type: bigquery.StringFieldType},
+			}},
+	}
+
+	metaData := &bigquery.TableMetadata{
+		Schema:         sampleSchema,
+		ExpirationTime: time.Now().AddDate(1, 0, 0), // Table will be automatically deleted in 1 year.
```
Comment (tswast) [inline bigquery/snippets/snippet.go]: Is the expiration necessary for this sample? It seems unrelated to the actual schema demonstration.

File: bigquery/snippets/snippet.go [hunk 3]
```diff
@@ -407,6 +453,194 @@ func createTableExplicitSchema(client *bigquery.Client, datasetID, tableID strin
...
+	// Iterate through the schema to set all Required fields to false (nullable).
+	var relaxed bigquery.Schema
+	for _, v := range original.Schema {
+		v.Required = false
+		relaxed = append(relaxed, v)
+	}
+	newMeta := bigquery.TableMetadataToUpdate{
+		Schema: relaxed,
+	}
+	if _, err := tableRef.Update(ctx, newMeta, ""); err != nil {
```
Comment (tswast) [inline bigquery/snippets/snippet.go]: This seems like a prime example were we would want to use the etag, right? We're doing a get-modify-update.

File: bigquery/snippets/snippet.go [hunk 4]
```diff
@@ -407,6 +453,194 @@ func createTableExplicitSchema(client *bigquery.Client, datasetID, tableID strin
...
+	// [START bigquery_relax_column]
+	sampleSchema := bigquery.Schema{
+		{Name: "full_name", Type: bigquery.StringFieldType, Required: true},
+		{Name: "age", Type: bigquery.IntegerFieldType, Required: true},
+	}
+	original := &bigquery.TableMetadata{
+		Schema: sampleSchema,
+	}
+	tableRef := client.Dataset(datasetID).Table(tableID)
+	if err := tableRef.Create(ctx, original); err != nil {
```
Comment (tswast) [inline bigquery/snippets/snippet.go]: A table Get might be more natural for this sample (with the create outside the sample).

File: bigquery/snippets/snippet.go [hunk 5]
```diff
@@ -1136,6 +1474,134 @@ func importJSONAutodetectSchema(client *bigquery.Client, datasetID, tableID stri
...
+	if status.Err() != nil {
+		return fmt.Errorf("Job completed with error: %v", status.Err())
+	}
+	// [END bigquery_load_table_gcs_orc]
+	return nil
+}
+
+func importORCTruncate(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// [START bigquery_load_table_gcs_orc_truncate]
```
Comment (tswast) [inline bigquery/snippets/snippet.go]: Do we want to combo all the `_append` and `_truncate` samples like we did in Python to save on system test time?

https://github.com/GoogleCloudPlatform/google-cloud-python/blob/b176b888d56651689bea8768d39e57e518660ce7/docs/bigquery/snippets.py#L1543-L1559

File: bigquery/snippets/snippet.go (hunk 3 above)
Comment (shollyman) [reply bigquery/snippets/snippet.go]: Potentially, but getting an Etag would require another roundtrip of metadata fetch, since Go doesn't return the metadata on create (only any potential errors).

Schema evolution is more restrictive in that it only allows "compatible" evolutions of the schema, so the modify would fail regardless if the new schema is incompatible (e.g. an intermediate modify added a column).  Perhaps a comment rather than another fetch?

File: bigquery/snippets/snippet.go (hunk 2 above)
Comment (shollyman) [reply bigquery/snippets/snippet.go]: Will yank it.

File: bigquery/snippets/snippet.go (hunk 4 above)
Comment (shollyman) [reply bigquery/snippets/snippet.go]: It would simplify the other point about schema evolution, will do that instead.

Comment (shollyman) [reply bigquery/snippets/snippet.go]: s/that/this/

File: bigquery/snippets/snippet.go (hunk 5 above)
Comment (shollyman) [reply bigquery/snippets/snippet.go]: My primary question is how are people consuming the samples?  If they're consuming from the docs, then the sample is straightforward.  If they're using the "view on github", then tag interleaving within a method like that makes for somewhat confusing samples.

File: bigquery/snippets/snippet.go (hunk 3 above)
Comment (tswast) [reply bigquery/snippets/snippet.go]: > Perhaps a comment rather than another fetch?

SGTM

File: bigquery/snippets/snippet.go (hunk 5 above)
Comment (tswast) [reply bigquery/snippets/snippet.go]: Our hypothesis is copy-paste from cloud.google.com, but it's a good question.

File: bigquery/snippets/snippet.go [hunk 6]
```diff
@@ -457,6 +694,47 @@ func createTableClustered(client *bigquery.Client, datasetID, tableID string) er
...
+	ctx := context.Background()
+	// [START bigquery_update_table_cmek]
+	tableRef := client.Dataset(datasetID).Table(tableID)
+	meta, err := tableRef.Metadata(ctx)
+	if err != nil {
+		return err
+	}
+	update := bigquery.TableMetadataToUpdate{
+		EncryptionConfig: &bigquery.EncryptionConfig{
+			KMSKeyName: "projects/cloud-samples-tests/locations/us-central1/keyRings/test/cryptoKeys/otherkey",
```
Comment (tbpg) [inline bigquery/snippets/snippet.go]: Add TODO line like the other samples?

File: bigquery/snippets/snippet.go [hunk 7]
```diff
@@ -1136,6 +1477,134 @@ func importJSONAutodetectSchema(client *bigquery.Client, datasetID, tableID stri
...
+	return nil
+}
+
+func importORCTruncate(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// [START bigquery_load_table_gcs_orc_truncate]
+	gcsRef := bigquery.NewGCSReference("gs://cloud-samples-data/bigquery/us-states/us-states.orc")
+	gcsRef.SourceFormat = bigquery.ORC
+	loader := client.Dataset(datasetID).Table(tableID).LoaderFrom(gcsRef)
+	// default for import jobs is to append data to a table.  WriteTruncate
```
Comment (tbpg) [inline bigquery/snippets/snippet.go]: Nit: Capitalize.
